### PR TITLE
Fix and add benchs for democracy pallet

### PR DESCRIPTION
**Pull request by thiolliere**
* external propose now consider the cost of a blacklist with multiple vetoers
* add a benchmark for on_initialize when nothing happens (no maturing referenda nor launch)

### CODE REVIEW

File: frame/democracy/src/benchmarking.rs [hunk 1]
```diff
@@ -467,6 +471,36 @@ benchmarks! {
 		}
 	}
 
+	on_initialize_no_launch_no_maturing {
```
Comment (kianenigma) [inline frame/democracy/src/benchmarking.rs:474]: how is this different from all the other `on_initialize_xxx` benches? would probably be helpful to write it down here.

Comment (thiolliere) [reply frame/democracy/src/benchmarking.rs:474]: yes on_initialize_external do benchmark the launch of an external referendum and the maturing of R referendums.
and on_initialize_public does same with public referendum.

I agree doc should improve here, I'll try to do in my weight PR

File: frame/democracy/src/benchmarking.rs [hunk 2]
```diff
@@ -296,13 +299,14 @@ benchmarks! {
...
 		let p in 1 .. MAX_PROPOSALS;
+		let v in 1 .. MAX_VETOERS;
 
 		let origin = T::ExternalOrigin::successful_origin();
 		let proposal_hash = T::Hashing::hash_of(&p);
 		// Add proposal to blacklist with block number 0
 		Blacklist::<T>::insert(
 			proposal_hash,
-			(T::BlockNumber::zero(), vec![T::AccountId::default()])
+			(T::BlockNumber::zero(), (0..v).map(|_| T::AccountId::default()).collect::<Vec<_>>())
```
Comment (apopiak) [inline frame/democracy/src/benchmarking.rs:309]: shouldn't `vec![T::AccountId::default(); v]` work as well?
seems a bit simpler
```suggestion
			(T::BlockNumber::zero(), vec![T::AccountId::default(); v])
```

**Comment by thiolliere** [comment]
bot merge

**Comment by thiolliere** [comment]
bot merge

**Pull request merged**